Flint/src/main/java/flint/FlintDriver.java

```diff
@@ -1,10 +1,51 @@
 package flint;
 
 import java.util.Collection;
+import javax.tools.*;
+
 
 public class FlintDriver {
-  public static Collection<LintFailure> run(String fileToLint, FlintConfiguration config) {
-    System.out.println("Not yet implemented!");
-    return null;
+  /**
+   * This method checks whether fileToLint refers to a compileable Java class,
+   * throwing an IllegalArgumentException if not, then produces a Scanner to the
+   * file as well as a CompilationUnit (the result of parsing the file and the
+   * root of an Abstract Syntax Tree), which it passes to config.
+   * 
+   * @param fileToLint - Filepath to the file we wish to lint
+   * @param config     - Configuration file that defines how to run the checks
+   * @return Every LintFailure resulting from running the checks in config
+   * @throws FileNotFoundException    - If fileToLint does not refer to a valid
+   *                                  file
+   * @throws IllegalArgumentException - If fileToLint does not compile
+   * @throws ParseProblemException    - If fileToLint could not be successfully
+   *                                  parsed
+   */
+  public static Collection<LintFailure> run(String fileToLint, FlintConfiguration config)
+      throws FileNotFoundException, IllegalArgumentException, ParseProblemException, IOException {
+    if (doesCompile(fileToLint)) {
+      RandomAccessFile inputFile = new RandomAccessFile(fileToLint, "r");
+      Collection<LintFailure> result = config.runChecks(inputFile, JavaParser.parse(new File(fileToLint)));
```

Comment:
@jessica60933 you might want to talk with @elliottdebruin about what kind of parameters should `runChecks()` take, since I didn't see any parameter specified in the spec of FlintConfiguration

Reply:
My thought on Config ADT is that it will have the following fields
```
  CompilationUnit astRoot;
  RandomAccessFile inputFile;
  Collection<FlintRule> rules;
  Collection<LintFailure> failures;
```
and the non-static method signature `public abstract Collection<LintFailure> runChecks();`

Does this make sense?